tests/core/index_management/test_index_management.py

```diff
@@ -276,3 +279,85 @@ def test_get_marqo_version_v20_successful(self):
             )
 
             self.assertEqual(self.index_management.get_marqo_version(), '2.0')
+
+    def test_createAndDeleteIndexCannotBeConcurrent(self):
+        """Test to ensure create_index and delete_index is not concurrent"""
+        index_name_1 = 'a' + str(uuid.uuid4()).replace('-', '')
+        marqo_index_request_1 = self.structured_marqo_index_request(
+            name=index_name_1,
+            model=Model(name='ViT-B/32'),
+            distance_metric=DistanceMetric.PrenormalizedAngular,
+            vector_numeric_type=VectorNumericType.Float,
+            hnsw_config=HnswConfig(ef_construction=100, m=16),
+            fields=[
+                FieldRequest(name='title', type=FieldType.Text, features=[FieldFeature.LexicalSearch]),
+            ],
+            tensor_fields=[]
+        )
+
+        index_name_2 = 'a' + str(uuid.uuid4()).replace('-', '')
+        marqo_index_request_2 = self.structured_marqo_index_request(
+            name=index_name_2,
+            model=Model(name='ViT-B/32'),
+            distance_metric=DistanceMetric.PrenormalizedAngular,
+            vector_numeric_type=VectorNumericType.Float,
+            hnsw_config=HnswConfig(ef_construction=100, m=16),
+            fields=[
+                FieldRequest(name='title', type=FieldType.Text, features=[FieldFeature.LexicalSearch]),
+            ],
+            tensor_fields=[]
+        )
+
+        def create_index(marqo_index_request):
+            self.index_management.create_index(marqo_index_request)
+
+        t_1 = threading.Thread(target=create_index, args=(marqo_index_request_1,))
+        t_1.start()
+        time.sleep(1)
+        with self.assertRaises(IndexCreationAndDeletionConflictError):
+            self.index_management.create_index(marqo_index_request_2)
+
+        with self.assertRaises(IndexCreationAndDeletionConflictError):
+            self.index_management.delete_index_by_name(index_name_1)
+        t_1.join()
+        self.index_management.delete_index_by_name(index_name_1)
+
+    def test_createIndexFailIfNoZookeeperProvided(self):
+        self.index_management = IndexManagement(self.vespa_client, zookeeper_client=None)
+        index_name = 'a' + str(uuid.uuid4()).replace('-', '')
+        marqo_index_request = self.structured_marqo_index_request(
+            name=index_name,
+            model=Model(name='ViT-B/32'),
+            distance_metric=DistanceMetric.PrenormalizedAngular,
```

Comment:
Why do you need all this custom index config for this test? Seems irrelevant to the test case

Reply:
use an unstructured index with defaults to make it easier